content/400-reference/200-api-reference/250-error-reference.mdx

```diff
@@ -181,6 +136,14 @@ Possible errors:
 
 "Server has closed the connection."
 
+#### <inlinecode>P1018</inlinecode>
+
+"{message}"
+
+#### <inlinecode>P1019</inlinecode>
+
+"{message}"
```

Comment:
Script might not be capable of getting actual message here, or it is so variable as P1012 that it would make sense to get the actual, additional error codes either manually or adapting the script.

Reply:
I'm not sure where to get the errors from. Looking at the script, for P1012 it gets the list of messages from [here](https://raw.githubusercontent.com/prisma/prisma-engines/main/libs/datamodel/diagnostics/src/error.rs) (the updated path for the `url5` link that returns a 404). But it's not obvious to me why these were used for P1012, or where I'd get the corresponding list for P1018 and P1019.

Reply:
All the errors are in these files https://github.com/prisma/prisma-engines/tree/main/libs/user-facing-errors/src

Reply:
The engine code is not necessarily written to make it easy to parse this way, so Martina (who built this script) probably just asked and then added a special case for that file where it made sense - possibly the file that @garrensmith linked to might be the correct one to use here (or manually of course).

Side note: Adding to the file which section was auto generated and which one edited manually might be useful for the future.